Foundation/Date Range/TWTDateRange.m

```diff
@@ -0,0 +1,130 @@
+//
+//  TWTDateRange.m
+//  Toast
+//
+//  Created by Duncan Lewis on 3/23/2015.
+//  Copyright (c) 2015 Ticketmaster Entertainment, Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "TWTDateRange.h"
+
+
+@implementation TWTDateRange
+
+- (instancetype)init
+{
+    return [self initWithStartDate:nil endDate:nil];
+}
+
+
+- (instancetype)initWithStartDate:(NSDate *)startDate endDate:(NSDate *)endDate
+{
+    self = [super init];
+    if (self) {
+        _startDate = startDate ? startDate : [NSDate distantPast];
+        _endDate = endDate ? endDate : [NSDate distantFuture];
+        NSAssert([startDate compare:endDate] <= NSOrderedSame, @"The start date must not occur after the end date");
```

Comment:
Since we're documenting that this assertion is raised on `@param endDate ...`, maybe the wording of this assertion should be "The end date (%@) must not occur before the start date (%@)." and supply the description of the dates